### ISSUE TITLE
Support for chained / nested watch

### ISSUE THREAD

**Issue by simontaurus** [json-editor/json-editor#1532]
### General information

- json-editor version: [2.14.1](https://github.com/json-editor/json-editor/releases/tag/2.14.1)

### Expected behavior

Watching another watching field should work in the way that an updates propagates through the chain of watching fields,
e. g. FieldA -watched by> FieldB -watched by> FieldC
as long as there are no circular dependancies.

### Actual behavior

Change event stops at first watching element in the chain, here FieldB

Workaround: 
Instead of building a chain FieldA -> FieldB -> FieldC,
FieldA -> FieldB and FieldA ->  FieldC can be used since both FieldB and FieldC have template strings that can be merged, so thats maybe just a documentation issue.

### Steps to reproduce the behavior

Direct link to example: [playground](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUBXRBTA+vAhgLa75RQBOMARulFsihelgDQgTliHQ5VYBmYcrgAmWADZY6qJqxCZRA/OnFQcAN3zjmDWW0L4AHjjHQAFqgAMbGmQQ5EUUrkJgxqAIxsAxvmwOsJBhYdVwOSCxyKABPAPxybwtGcmY2RDMwAHccSM5yBhAYeDpyfG9YBBA2GG8EcWpUEEFi/Ez6LiwAVirwKgArLHKccXxosFpG+CEDcR7ERKwDVFAYiCxGsH7BqB7YKElGgBkwXwr4HpgRRs4wHbZwtaiYemWQb2Do19X1lBBHSngAHMemJ+MpVI0AMr4eAAAgAYqV4O95mAAEwgAC+aSc0jQIG+jX+RWBbFB4J2vwAwgBBLE+D6OZxfaJrIkUEkg+jeSjQODnX4AFTMMEQsNFsMBgUizhEsOUUC4pBqWnE0Vh/E4hFhUDMWFhHCw6jgmB1mTAGue4hEyDYdG4IzxIGAwHeMUx2NhLqZdA9PUypESrzdn1+NygADoQ3NcT8QOGIz71h6GTEkxj8YTfsSgVz5ryzo1hRKJVL4DK6HKFUrYL5xGqNVqdXqDcJjeMxVBzZaJDbdosII64y6Q0mPTr6HcQAGoEH8aPY9cwLco4zFx6UwS9cRGlRl1B/vgIAAWLGYoA)

```javascript
{
  "type": "object",
  "title": "Location",
  "id": "root",
  "properties": {
    "city": {
      "type": "string",
      "default": "San Francisco2"
    },
    "state": {
      "type": "string",
      "default": "CA"
    },
    "citystate": {
      "type": "string",
      "description": "This is generated automatically from the previous two fields",
      "template": "{{city}}, {{state}}",
      "watch": {
        "city": "root.city",
        "state": "root.state"
      }
    },
    "citystate2": {
      "type": "string",
      "description": "This is generated automatically from the previous field",
      "template": "{{citystate}} test",
      "watch": {
        "citystate": "root.citystate"
      }
    }
  }
}
```

**Comment by schmunk42** [comment]
> as long as there are no circular dependancies.

This might be hard to detect. @germanbisurgi WDYT?

But a line in the docs about stopping at the first element would be good.

**Comment by germanbisurgi** [comment]
Fix in #1547
